src/box/sql/build.c

```diff
@@ -558,7 +552,7 @@ field_def_create_for_pk(struct Parse *parser, struct field_def *field,
 void
 sqlAddPrimaryKey(struct Parse *pParse)
 {
-	struct space *space = pParse->create_column_def.space;
+	struct space *space = pParse->space;
```

Comment:
I stand by what I said in another comment. I think this is wrong to merge such generic members back into Parse/sql_parse. This is already problematic to read, when I simply don't understand who uses `Parse.space`. Previously it was clear from the sub-struct names like `create_column_def`. IMO would be better to keep separate struct per statement type. And maybe even have some members duplicated in them, but by looking at each such struct I would immediately understand what data it needs.

As an example you can look at `xrow.h` and its structs `register_request`, `subscribe_request`, `subscribe_response`, `join_request` and others. They have a few duplicate members, but in each request it is crystal clear what it operates with.

Try to open `struct Parse` and tell which members are going to be used by `ALTER TABLE ADD COLUMN`. Among these: `autoname_i`, `triggered_space`, `column_list`, `foreign_key_list`, `check_list`, `unique_list`, `primary_key`, `create_index`, `autoinc_name`, `src_list`, `space`, `has_autoinc`, `autoinc_fieldno`? Personally, I have no idea ATM. You probably know because you work with this right now, but ask anybody else and they wouldn't be as confused as me. I know some of these members were here before this patchset, but now it feels we are loosing control again.